packages/react-reconciler/src/ReactFiberEvents.js

```diff
@@ -165,9 +177,11 @@ export function updateEventListeners(
     if (respondersMap === null) {
       respondersMap = new Map();
     }
+
     if (isArray(listeners)) {
-      for (let i = 0, length = listeners.length; i < length; i++) {
-        const listener = listeners[i];
+      const flatListeners = flatten(listeners, []);
```

Comment:
We're doing this regardless of the listeners not needing to be flattened? This is not ideal.